contracts/eth/nearbridge/test/NearBridge3.js

```diff
@@ -53,8 +62,119 @@ describe('NearBridge with admin access', () => {
         );
     });
 
-    describe('Upgradability', async () => {
-        it('should upgrade the admin address from the provided hex string', async () => {
+    describe('AdminControlled', async () => {
+        it('Admin account matches', async() => {
+            expect(
+                await nearBridge.admin()
+            )
+                .to
+                .be
+                .equal(adminAccount.address);
+        });
+
+        it('regular user can not perform admin functions', async() => {
+            const recipientBalanceBefore = ethers.BigNumber.from(await ethers.provider.getBalance(userAccount2.address));
+            const contractBalanceBefore = ethers.BigNumber.from(await ethers.provider.getBalance(nearBridge.address));
+
+            const amountToTransfer = 4000; // wei
+            // user1 tries to perform `adminSendEth()` to replenish user2 balance
+            await expect(
+                nearBridge
+                    .connect(userAccount1)
+                    .adminSendEth(userAccount2.address, amountToTransfer)
+            )
+                .to
+                .be
+                .reverted;
+
+            const recipientBalanceAfter = ethers.BigNumber.from(await ethers.provider.getBalance(userAccount2.address));
+            const contractBalanceAfter = ethers.BigNumber.from(await ethers.provider.getBalance(nearBridge.address));
+
+            expect(recipientBalanceAfter)
+                .to
+                .be
+                .equal(recipientBalanceBefore);
+            expect(contractBalanceAfter)
+                .to
+                .be
+                .equal(contractBalanceBefore);
+
+            // Try to pause and unpause
+            await expect(nearBridge.connect(userAccount1).adminPause(PAUSED_DEPOSIT)).to.be.reverted;
+            await expect(nearBridge.connect(userAccount1).adminPause(PAUSED_WITHDRAW)).to.be.reverted;
+            await expect(nearBridge.connect(userAccount1).adminPause(PAUSED_ADD_BLOCK)).to.be.reverted;
+            await expect(nearBridge.connect(userAccount1).adminPause(PAUSED_CHALLENGE)).to.be.reverted;
+            await expect(nearBridge.connect(userAccount1).adminPause(PAUSED_VERIFY)).to.be.reverted;
+            await expect(nearBridge.connect(userAccount1).adminPause(UNPAUSED_ALL)).to.be.reverted;
+            // ------------------------------------------
+
+            // Try to use adminSstore
+            await expect(
+                nearBridge
+                    .connect(userAccount1)
+                    .adminSstore(0, 1)
```

Comment:
what are `0` and `1` referring to ?

Reply:
These are just dummy values, in this specific call, it's not really important as we are checking this will be reverted.